tests/test_campaign.py

```diff
@@ -29,3 +35,32 @@ def test_get_surrogate(campaign, n_iterations, batch_size):
 
     model = campaign.get_surrogate()
     assert model is not None, "Something went wrong during surrogate model extraction."
+
+
+@pytest.mark.parametrize(
+    ("anti", "expected"),
+    [
+        (
+            False,
+            pd.DataFrame(columns=["a", "b"], data=[[0.0, 3.0], [0.0, 4.0], [0.0, 5.0]]),
+        ),
+        (
+            True,
+            pd.DataFrame(columns=["a", "b"], data=[[1.0, 3.0], [1.0, 4.0], [1.0, 5.0]]),
+        ),
+    ],
+    ids=["regular", "anti"],
+)
+def test_candidate_filter(anti, expected):
+    """The candidate filter extracts the correct subset of points."""
+    subspace = SubspaceDiscrete.from_product(
+        [
+            NumericalDiscreteParameter("a", [0, 1]),
+            NumericalDiscreteParameter("b", [3, 4, 5]),
+        ]
+    )
+    campaign = Campaign(subspace)
+    df = campaign.toggle_discrete_candidates(pd.DataFrame({"a": [0]}), False, anti=anti)
+    assert_frame_equal(
+        df, pd.merge(df.reset_index(), expected).set_index("index"), check_names=False
```

Comment:
just making sure: this merge is to make it invariant to the row order?

Reply:
Yes, had it hardcoded earlier but this is how I noticed #424. The workaround is pretty ugly compared to the original `assert_frame_equal(df, expected)`, though 😢 Would appreciate to hear cleverer ideas.

Reply:
its actually ridiculous that assert_frame_equal has no such option to allow row/column order invariance
pandas contribution?

Reply:
There actually is something in that direction: it takes a `check_like` flag which disables the ordering check. However, it doesn't really help here since it still requires fixed assignments of content to index/columns. So if you only want to "match content" (i.e. without column / index information), there is no solution AFAIK

So due to the lack of better alternatives (and since it's just a test, not production code), let me resolve